app/controllers/api/v1/admin/tags_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class Api::V1::Admin::TagsController < Api::BaseController
+  include Authorization
+  before_action -> { authorize_if_got_token! :'admin:read' }, only: [:index, :show]
+  before_action -> { authorize_if_got_token! :'admin:write' }, only: :update
+
+  before_action :set_tags, only: :index
+  before_action :set_tag, except: :index
+
+  after_action :insert_pagination_headers, only: :index
+  after_action :verify_authorized
+
+  LIMIT = 100
+  PAGINATION_PARAMS = %i(limit).freeze
+
+  def index
+    authorize :tag, :index?
+    render json: @tags, each_serializer: REST::Admin::TagSerializer
+  end
+
+  def show
+    authorize @tag, :show?
+    render json: @tag, serializer: REST::Admin::TagSerializer
+  end
+
+  def update
+    authorize @tag, :update?
+    if @tag.update(tag_params.merge(reviewed_at: Time.now.utc))
+      render json: @tag, serializer: REST::Admin::TagSerializer
+    else
+      @tag.reload
+      render json: { error: I18n.t('tags.does_not_match_previous_name') }, status: 422 unless tag_params[:display_name].casecmp(@tag.name.mb_chars).zero?
+    end
```

Comment:
This won't display anything in the event the reason for the failure is different.

I would just do `@tag.update!(…)` and let the error handling (from `app/controllers/api/base_controller.rb`) catch the exception and return the appropriate error:
```suggestion
    @tag.update!(tag_params.merge(reviewed_at: Time.now.utc))
```